File: app/core/message_status.py
"""
消息状态定义
扩展为7种细分状态，提供更精确的状态管理

"""
from enum import Enum
from typing import List


class MessageStatus(Enum):
    """
    消息状态 - 7种细分状态
    """
    # 待处理状态
    PENDING = "pending"              # 待审核 - 未发送失败过的消息
    SEND_FAILED = "send_failed"      # 发送失败 - 发送失败的消息

    # 已发布状态
    AUTO_APPROVED = "auto_approved"    # 自动发布 - 系统自动发布的消息
    MANUAL_APPROVED = "manual_approved" # 手动发布 - 人工发布的消息

    # 已拒绝状态
    AD_REJECTED = "ad_rejected"         # 广告拒绝 - 检测为广告被拒绝
    DUP_REJECTED = "dup_rejected"       # 重复拒绝 - 检测为重复被拒绝
    MANUAL_REJECTED = "manual_rejected" # 手动拒绝 - 人工拒绝

    @classmethod
    def get_approved_statuses(cls) -> List[str]:
        """获取所有已发布状态"""
        return [cls.AUTO_APPROVED.value, cls.MANUAL_APPROVED.value]

    @classmethod
    def get_rejected_statuses(cls) -> List[str]:
        """获取所有已拒绝状态"""
        return [cls.AD_REJECTED.value, cls.DUP_REJECTED.value, cls.MANUAL_REJECTED.value]

    @classmethod
    def get_pending_like_statuses(cls) -> List[str]:
        """获取所有待处理状态（包括待审核和发送失败）"""
        return [cls.PENDING.value, cls.SEND_FAILED.value]


# 向后兼容：旧的3状态系统
LEGACY_PENDING = "pending"
LEGACY_APPROVED = "approved"
LEGACY_REJECTED = "rejected"


class RejectionReason(Enum):
    """
    拒绝原因 - 保留用于向后兼容
    """
    AD = "ad"                # 广告内容
    DUP = "dup"              # 重复内容
    MANUAL = "manual"        # 手动拒绝

def is_valid_status(status: str) -> bool:
    """检查状态是否有效（支持新7状态和旧3状态）"""
    # 检查新的7状态
    try:
        MessageStatus(status)
        return True
    except ValueError:
        # 兼容旧的3状态
        return status in [LEGACY_PENDING, LEGACY_APPROVED, LEGACY_REJECTED]

def is_valid_rejection_reason(reason: str) -> bool:
    """检查拒绝原因是否有效"""
    try:
        RejectionReason(reason)
        return True
    except ValueError:
        return False


def get_status_display_name(status: str) -> str:
    """获取状态的显示名称"""
    display_names = {
        MessageStatus.PENDING.value: "待审核",
        MessageStatus.SEND_FAILED.value: "发送失败",
        MessageStatus.AUTO_APPROVED.value: "自动发布",
        MessageStatus.MANUAL_APPROVED.value: "手动发布",
        MessageStatus.AD_REJECTED.value: "广告拒绝",
        MessageStatus.DUP_REJECTED.value: "重复拒绝",
        MessageStatus.MANUAL_REJECTED.value: "手动拒绝",
        # 向后兼容
        LEGACY_APPROVED: "已发布",
        LEGACY_REJECTED: "已拒绝"
    }
    return display_names.get(status, status)


def get_rejection_reason_display_name(reason: RejectionReason) -> str:
    """获取拒绝原因的显示名称"""
    display_names = {
        RejectionReason.AD: "广告内容",
        RejectionReason.DUP: "重复内容",
        RejectionReason.MANUAL: "手动拒绝"
    }
    return display_names.get(reason, reason.value)


def map_legacy_status(status: str, reject_reason: str = None) -> str:
    """
    将旧的3状态系统映射到新的7状态系统

    Args:
        status: 旧状态值 (pending/approved/rejected)
        reject_reason: 拒绝原因（用于细分rejected状态）

    Returns:
        新状态值
    """
    if status == LEGACY_PENDING:
        return MessageStatus.PENDING.value
    elif status == LEGACY_APPROVED:
        return MessageStatus.MANUAL_APPROVED.value
    elif status == LEGACY_REJECTED:
        if reject_reason:
            if "广告" in reject_reason or "ad" in reject_reason.lower():
                return MessageStatus.AD_REJECTED.value
            elif "重复" in reject_reason or "dup" in reject_reason.lower():
                return MessageStatus.DUP_REJECTED.value
        return MessageStatus.MANUAL_REJECTED.value
    else:
        # 已经是新状态，直接返回
        return status


def get_legacy_status(status: str) -> str:
    """
    将新的7状态映射回旧的3状态（向后兼容）

    Args:
        status: 新状态值

    Returns:
        旧状态值 (pending/approved/rejected)
    """
    if status in MessageStatus.get_pending_like_statuses():
        return LEGACY_PENDING
    elif status in MessageStatus.get_approved_statuses():
        return LEGACY_APPROVED
    elif status in MessageStatus.get_rejected_statuses():
        return LEGACY_REJECTED
    else:
        # 未知状态，返回原值
        return status

File: static/assets/js/config/message-status.js
// 消息状态定义 - 与后端保持一致

window.MessageStatus = {
    // 7种细分状态
    PENDING: 'pending',               // 待审核 - 未发送失败过
    SEND_FAILED: 'send_failed',       // 发送失败
    AUTO_APPROVED: 'auto_approved',   // 自动发布
    MANUAL_APPROVED: 'manual_approved',// 手动发布
    AD_REJECTED: 'ad_rejected',       // 广告拒绝
    DUP_REJECTED: 'dup_rejected',     // 重复拒绝
    MANUAL_REJECTED: 'manual_rejected',// 手动拒绝

    // 兼容旧3状态系统
    LEGACY_PENDING: 'pending',
    LEGACY_APPROVED: 'approved',
    LEGACY_REJECTED: 'rejected',

    // 状态显示信息
    getStatusInfo(status) {
        const statusInfo = {
            // 新状态
            'pending': {
                label: '待审核',
                color: '#409EFF',
                bgColor: 'rgba(64, 158, 255, 0.1)',
                icon: 'clock',
                description: '等待审核的消息'
            },
            'send_failed': {
                label: '发送失败',
                color: '#F56C6C',
                bgColor: 'rgba(245, 108, 108, 0.1)',
                icon: 'exclamation-triangle',
                description: '发送失败需要重试'
            },
            'auto_approved': {
                label: '自动发布',
                color: '#67C23A',
                bgColor: 'rgba(103, 194, 58, 0.1)',
                icon: 'check-circle',
                description: '系统自动发布'
            },
            'manual_approved': {
                label: '手动发布',
                color: '#19A77B',
                bgColor: 'rgba(25, 167, 123, 0.1)',
                icon: 'user-check',
                description: '人工审核发布'
            },
            'ad_rejected': {
                label: '广告拒绝',
                color: '#E6A23C',
                bgColor: 'rgba(230, 162, 60, 0.1)',
                icon: 'ban',
                description: '检测为广告拒绝'
            },
            'dup_rejected': {
                label: '重复拒绝',
                color: '#909399',
                bgColor: 'rgba(144, 147, 153, 0.1)',
                icon: 'copy',
                description: '检测为重复拒绝'
            },
            'manual_rejected': {
                label: '手动拒绝',
                color: '#F56C6C',
                bgColor: 'rgba(245, 108, 108, 0.1)',
                icon: 'times-circle',
                description: '人工审核拒绝'
            },
            // 兼容旧状态
            'approved': {
                label: '已发布',
                color: '#67C23A',
                bgColor: 'rgba(103, 194, 58, 0.1)',
                icon: 'check-circle',
                description: '消息已发布'
            },
            'rejected': {
                label: '已拒绝',
                color: '#F56C6C',
                bgColor: 'rgba(245, 108, 108, 0.1)',
                icon: 'times-circle',
                description: '消息已拒绝'
            }
        };

        return statusInfo[status] || {
            label: status,
            color: '#909399',
            bgColor: 'rgba(144, 147, 153, 0.1)',
            icon: 'question-circle',
            description: '未知状态'
        };
    },

    // 判断是否为已发布状态
    isApproved(status) {
        return [this.AUTO_APPROVED, this.MANUAL_APPROVED, this.LEGACY_APPROVED].includes(status);
    },

    // 判断是否为已拒绝状态
    isRejected(status) {
        return [this.AD_REJECTED, this.DUP_REJECTED, this.MANUAL_REJECTED, this.LEGACY_REJECTED].includes(status);
    },

    // 判断是否为待处理状态
    isPending(status) {
        return [this.PENDING, this.SEND_FAILED].includes(status);
    },

    // 获取旧状态（向后兼容）
    getLegacyStatus(status) {
        if (this.isPending(status)) {
            return this.LEGACY_PENDING;
        } else if (this.isApproved(status)) {
            return this.LEGACY_APPROVED;
        } else if (this.isRejected(status)) {
            return this.LEGACY_REJECTED;
        }
        return status;
    }
};